lib/config.js

```diff
@@ -107,16 +106,17 @@ function hasRules(options) {
  * @returns {Object} The local config object, or an empty object if there is no local config.
  */
 function getLocalConfig(thisConfig, directory) {
-    const localConfigFiles = thisConfig.findLocalConfigFiles(directory),
-        numFiles = localConfigFiles.length,
-        projectConfigPath = ConfigFile.getFilenameForDirectory(thisConfig.options.cwd);
+
+    const projectConfigPath = ConfigFile.getFilenameForDirectory(thisConfig.options.cwd);
+    const localConfigFiles = thisConfig.findLocalConfigFiles(directory);
     let found,
         config = {},
         rootPath;
```

Comment:
Since `break` is now used when a config file with `root` is found, I think [this statement](https://github.com/VictorHom/eslint/blob/55ac32ac3fc5b3cfcdd9a775d682d6f97bafb60c/lib/config.js#L127-L130) can be removed.